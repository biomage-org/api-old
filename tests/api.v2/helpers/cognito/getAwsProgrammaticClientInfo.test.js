
const config = require('../../../../src/config');

const mockClients = {
  UserPoolClients: [
    {
      ClientId: '222222222222222222222222222',
      UserPoolId: 'eu-west-1_trivial',
      ClientName: 'biomage-cellscope-cluster-default',
    },
    {
      ClientId: '11111111111111111111111111',
      UserPoolId: 'eu-west-1_trivial',
      ClientName: 'biomage-programmatic-client-staging',
    },

  ],
};

jest.mock('../../../../src/config', () => ({
  ...jest.requireActual('../../../../src/config'),
  cognitoISP: {
    listUserPoolClients: jest.fn(() => ({
      promise: () => Promise.resolve(mockClients),
    })),
  },
}));

jest.mock('../../../../src/api.v2/helpers/cognito/getAwsPoolId');

const getAwsProgrammaticClientInfo = require('../../../../src/api.v2/helpers/cognito/getAwsProgrammaticClientInfo');


describe('getProgrammaticClientInfo', () => {
  it('returns a clientId and clientRegion when the client exists', async () => {
    const clientInfo = await getAwsProgrammaticClientInfo();
    expect(clientInfo).toMatchSnapshot();
  });

  it('returns a not found error when no matching client exists', async () => {
    config.cognitoISP.listUserPoolClients.mockImplementationOnce(jest.fn(() => ({
      promise: () => Promise.resolve({ UserPoolClients: [] }),
    })));
    try {
      await getAwsProgrammaticClientInfo();
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
