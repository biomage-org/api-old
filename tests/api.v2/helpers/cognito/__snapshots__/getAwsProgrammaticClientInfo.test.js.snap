// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProgrammaticClientInfo returns a clientId and clientRegion when the client exists 1`] = `
Object {
  "clientId": "11111111111111111111111111",
  "clientRegion": "eu-west-1",
}
`;

exports[`getProgrammaticClientInfo returns a not found error when no matching client exists 1`] = `[Error: getAwsProgrammaticClientInfo: cognito client biomage-programmatic-client-staging: not found]`;
